core/src/main/scala/kafka/admin/AdminClient.scala

```diff
@@ -49,7 +49,7 @@ class AdminClient(val time: Time,
 
     do {
       future = client.send(target, api, request)
-      client.poll(future)
+      client.poll(future, deadline - now)
```

Comment:
That's a very good idea. If we check that it has failed, we can guarantee a non-null exception. If it hasn't failed and hasn't succeeded, then we can safely throw a TimeoutException.